vizro-core/src/vizro/models/_navigation/nav_item.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import annotations
+
+import itertools
+from typing import Optional
+
+import dash
+import dash_bootstrap_components as dbc
+from dash import html
+from pydantic import Field, root_validator, validator
+
+from vizro.models import VizroBaseModel
+from vizro.models._models_utils import _log_call
+from vizro.models._navigation._navigation_utils import _validate_pages
+from vizro.models._navigation.accordion import Accordion
+from vizro.models.types import NavPagesType
+
+
+class NavItem(VizroBaseModel):
+    """Icon to be used in Navigation Panel of Dashboard.
+
+    Args:
+        pages (NavPagesType): See [NavPagesType][vizro.models.types.NavPagesType].
+        icon (str): URI (relative or absolute) of the embeddable content. Defaults to "home".
+        max_text_length (int): Character limit for `text` argument. Defaults to 8.
+        text (Optional[str]): Text to be displayed below the icon. Defaults to `None`.
+        tooltip (Optional[str]): Text to be displayed in the icon tooltip on hover.
+        selector (Optional[Accordion]): See [`Accordion`][vizro.models.Accordion]. Defaults to `None`.
```

Comment:
Would be great if you could update the docstrings to reflect e.g. that the text automatically gets concatenated based on max_text_length and that the full text is visible in the tooltip then etc.

The relationship between the 3 arguments should be clear